src/main/java/org/springframework/data/elasticsearch/client/reactive/RequestCreator.java

```diff
@@ -126,6 +126,10 @@
 		return RequestConverters::indexCreate;
 	}
 
+	default Function<org.elasticsearch.client.indices.CreateIndexRequest, Request> createIndex() {
```

Comment:
please add `@since 4.2`; and we should deprecate the `indexCreate()` method that uses the old class.
 It's a pity that we need a new name here. Although `createIndex()` is better than `indexCreate` in my opinion.
Or could use `createIndexRequest()` like you have it with the `putMappingRequest`, but then we probably should rename them all to `xxxRequest()`, deprecating the existing ones - might be something for a cleanup-ticket in the future.

But we should nevertheless deprecated the existing `indexCreate` method as it's only called by the deprectaed method from the client.